app.py

```diff
@@ -32,18 +30,27 @@ def __init__(self):
         else:
             self.drv = None
             self.enable_pin = None
+
         eventbus.on(ButtonDownEvent, self._handle_buttondown, self)
-        eventbus.on_async(HexpansionInsertionEvent, self.handle_hexpansion_insertion, self)
-        eventbus.on_async(HexpansionRemovalEvent, self.handle_hexpansion_removal, self)
+        eventbus.on(HexpansionInsertionEvent, self.handle_hexpansion_insertion, self)
```

Comment:
`eventbus.on` goes with synchronous methods (which you used). `eventbus.on_async` would require you to declare `handle_hexpansion_insertion` as an async method. They fail if you pass the wrong type.